[MDB IGNORE] A.S.S - Assault, Support, Security

Git-Nivrak opened the pull request:
# About the pull request
Changes up squads to have designated roles and purposes which also affect their role composition and size.

The three "types" of squads are as follows:

Assault (>=50% of squad marines):
Alpha and Delta, these will be the squads as you've known until today and will not be limited, for people looking to just play without extra responsibility. The only difference is that they will have two smartgunners now instead of one.

Support (<=30% of squad marines):
Charlie, Kilo and Oscar, these squads will serve as filling the niche for players looking for small, coordinated squad gameplay and serve as an unknown variable up to command's discretion. They have no smartgunner or team leader.

Security (<= 20 of squad marines):
Bravo, tasked with, you guessed it - ~~FOB~~ Security, This includes FOB, Comms and possibly other duties down the line with logistics changes. The specialist and smartgunner are removed from bravo.

![IMG_5456](https://github.com/user-attachments/assets/7b77f2ae-ff36-4715-a91a-892e851c8260)

<!-- Remove this text and explain what the purpose of your PR is.

Mention if you have tested your changes. If you changed a map, make sure you used the mapmerge tool.
If this is an Issue Correction, you can type "Fixes Issue #169420" to link the PR to the corresponding Issue number #169420.

Remember: something that is self-evident to you might not be to others. Explain your rationale fully, even if you feel it goes without saying. -->

# Explain why it's good for the game
ASS system introduces an asymmetrical squad structure that allows players to head into rounds under the pretense that they will be able to engage in a specific type of CM experience, while leaning into squad archetypes and giving more flexibility to command roles.

The Assault squads represent the classic, large group CM gameplay experience, where individual player agency is at a maximum and cohesion is a function of herd behavior.

Support squads introduce a more refined, guided experience where 3 small squads function at the behest of command (similar to a PvE style experience, and officers are giving more tactile control over a subset of players on the ground.

Security is the traditional slower-paced backline and defense group, streamlined to only contain roles relevant, while issuing specialist and smartgunner roles to other squads.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>


# Changelog
:cl:
refactor: All headsets are now self setting.
add: Kilo and Oscar squads.
balance: Increase SG slots on Alpha and Delta to 2, Reduced SG slots on rest of the squads to 0, Reduced specialist slots on Bravo to 0, Reduced FTL slots on Charlie, Oscar and Kilo to 0.
balance: Riflemen slots on Kilo, Oscar and Charlie will be limited to 10% of total squad marines each, 20% on Bravo
maptweak: Changed up squad preps into platoon preps, Platoon 1 is Alpha And Delta, Platoon 2 is Charlie, Oscar, Kilo and Bravo. Credit to thwomper.
/:cl:

Git-Nivrak committed:
init

Git-Nivrak committed:
tabs

Git-Nivrak committed:
beret desc

Git-Nivrak committed:
Fixes

Git-Nivrak committed:
Update squads.dm

Git-Nivrak committed:
Update squads.dm

Git-Nivrak committed:
a

Git-Nivrak committed:
adds ratio balancing, headsets are now self setting

Git-Nivrak committed:
fix

Git-Nivrak committed:
stages squad distribution, more fixes

Git-Nivrak committed:
Increase SG count

Git-Nivrak committed:
Linter stuff

Git-Nivrak committed:
Update ticker.dm

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Fixup maps in TGM format

b3bffaa072f5: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

cm13-github commented:
<!-- tgs_test_merge_comment -->
## Test merge deployment history:

<details><summary>Test Merge Deployed @ 09/13/2025 16:24:40 +00:00:</summary>

##### Server Instance

[CM13 Live](https://github.com/cmss13-devs/cmss13/deployments/activity_log?environment=TGS%3A+CM13+Live)

##### Revision
Origin: 198bf1e051ce642e2091a8e3f2089c005a4561aa
Pull Request: 71541aff64158807080c747c546feaec70fe105e
Server: 334659ba4c2bceb88119da87d54ae6c845e8ed58
</details>

cuberound commented:
made reprot, but extra squad slots like squad leader at least do not open correctly

Stinvals commented:
The Corporate Liaison's auto briefcase lacks the new comm keys (round 29894)
<img width="371" height="259" alt="image" src="https://github.com/user-attachments/assets/3382f77d-4661-4f57-b70f-c907051bffcb" />

Git-Nivrak committed:
Bunch of fixes

Git-Nivrak committed:
stripe sprites

Git-Nivrak committed:
Fixup maps in TGM format

b3bffaa072f5: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Adds missing SL slot

Fixes #10624

Git-Nivrak committed:
Six squad leaders, actually

Git-Nivrak committed:
Colors, More access fixes, Orbit menu and crew manifest, Encryption keys

Git-Nivrak committed:
Fix crew manifest ordering and colors

Git-Nivrak committed:
colon

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Update squads.dm

Git-Nivrak committed:
Colors

Git-Nivrak committed:
Background colors

SirNightKnight commented:
if you really wanna hone it in with squads having certain roles, maybe add the squad's role next to the squad's name in the job intro and squad preference in the character prefs sheet.

thwompa committed:
squad item missing icons

Git-Nivrak committed:
Merge pull request #24 from thwompa/ass-map-thwomping

missing icons

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Git-Nivrak committed:
Limits medic and ct to 1 on support squads

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Git-Nivrak committed:
color that exists

Git-Nivrak committed:
radio colors

Git-Nivrak committed:
Bunch of stuff (expand)

Removed specs from support squads, increased assault squad specs to two
Adds 1 TL to support squads

Git-Nivrak committed:
Fixes tracker, icon problems, adds squad designation information

Git-Nivrak committed:
edit the file on the actual repo this time

thwompa committed:
tango

thwompa committed:
map and landmarks

Git-Nivrak committed:
Expand

Fixes radio chat colors for real this time
Changed preferred squad into a priority system

Git-Nivrak committed:
Improves distribution, adds some lights

Git-Nivrak committed:
Fixup maps in TGM format

91e5e38769bd: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Priority input, Headset fixes

Git-Nivrak committed:
TGUI linters and tracker sprites

Git-Nivrak committed:
Update scarves.dmi

Git-Nivrak committed:
Kilo opens at 80 clients, Oscar at 120

Git-Nivrak committed:
typo

Git-Nivrak committed:
Unlocks squads based on pop

Git-Nivrak committed:
Fixes slots

Git-Nivrak committed:
remove debug messages

skypatrol21 commented:
This feel like a if it isnt broken dont fix it moment. While squads werent perfect this just makes them even more unappealing as a SL player. Alpha and Delta are too larged to ever be managed in a game likes this. A big squad size like that would work in a game with microphones but its hard to manage just 1 SG and 10 marines Doubling that is impossible. FTL need to be improved before squads this big would ever be fun.  This also has the issue with now doubling the amount of backline roles for marines.

Lukas-Seltz commented:
Big suggestion here make it so you can multicast to all 6 squads.... please

rocket2guns commented:
I think this change, in one sweep - destroys the organic "charlie lore" that has developed over a decade.

Git-Nivrak committed:
Expand

1. Readded colors to briefing and prep where possible
2. Added a centralized support overwatch console for Charlie, Kilo and Oscar
3. 2nd platoon SG drums moved to 1st platoon ammo vendors

Git-Nivrak committed:
Fixes access in ammo vendors

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Fixup maps in TGM format

91e5e38769bd: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

sanshoom commented:
I think this PR would be better if the support squads had better equipment.

Merocos commented:
I've sent in my PR Feedback, but my TLDR of ideas:

- Slightly larger squad for a total of 12; not 8. People still will always run off & 8 as a squad size even on mid/high pop is not conducive to an actual squad. That's closer to a Fireteam frankly and that's not a good thing. IE: 1 SL, 1 FTL, 2 Medics, 1 Engineer, 1 ...special (i.e some kind of 'focus' like engineering, recon, medical or combat support or just stick to K.I.S.S & go with a SG or Spec - I think SG is more balanced due to less killing power), 6 Pfcs
- Reduce 3 Support squads to 2: Charlie & Kilo: You're still keeping the 24 slots on 80sample - 12+12=24 slots vs 8+8+8=24 slots
- Platoon Channels for FTLs & SLs. Why? Comparable to JTAC. Can POs use command comms - sure. But it's good to have clear delineation if we want more refined leadership capabilities.

- Minor stuff (worth mentioning but not why I thumbs down): Squad prep is disgustingly bland & lacks character for squads that should have it.

I think it could work but if you told me about it as is I'd think it was just some silly meme

Git-Nivrak commented:
Reminder the place for feedback is on the forums https://forum.cm-ss13.com/w/pr-feedback/steps/step_1

Steelpoint reviewed (changes requested):
Some comments. 

- No security cameras in Alpha, Delta and Bravo cryo rooms (Admittedly old Almayer does not have them but support cryo does have one).
- Support Squad prep doors don't have a name telling you which prep rooms belongs to whom.
 - - Will also want to set the access locks to these doors to the correct jobs. 
 - I don't know if its intentional but the Delta briefing room has a bunch of Bravo chairs inside.
- Blue/teal chairs in shared support brief room could do being named to the squad.
- Platoon1_shared and Platoon2_shared areas should have an associated area icon, they have an error icon currently.

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Git-Nivrak committed:
Update squads.dm

Git-Nivrak committed:
fix

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Git-Nivrak committed:
Access stuff

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Git-Nivrak committed:
Remove vend x offset

Git-Nivrak committed:
Add missing trackers

Git-Nivrak committed:
Track LZ by default for SLs

Git-Nivrak committed:
Colors and chairs

Git-Nivrak committed:
Tacmap colors

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Makes it so you can log into both consoles

Git-Nivrak committed:
Increase pop locks a bit

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Allow setting primary and secondary through central overwatch

thwompa committed:
Merge branch 'ass3' of https://github.com/thwompa/cmss13 into ass3

thwompa committed:
tweaks oscar color for readability

thwompa committed:
oscar colors

thwompa committed:
floor tile sprite color fixes

Git-Nivrak committed:
Allows people to opt out of squads

Git-Nivrak committed:
Fix ghost roles taking up space

Git-Nivrak committed:
delete them as well afterwards

Git-Nivrak committed:
better

Git-Nivrak committed:
Merge pull request #32 from thwompa/colours

oscar color tweak

Git-Nivrak committed:
This one is for the CIC players

Git-Nivrak committed:
Brings IOs back from the dead

Git-Nivrak committed:
For real this time

Git-Nivrak committed:
prevent foxtrot edge cases

Git-Nivrak committed:
Fixes #10664

Git-Nivrak committed:
Fix #10670

Git-Nivrak committed:
Fixes #11364

Git-Nivrak committed:
Overwatch qol

Git-Nivrak committed:
No more 5x XO consoles

Git-Nivrak committed:
Possible fix to foxtrot spawn

Git-Nivrak committed:
Better fix hopefully?

thwompa committed:
canteen access and ow

Git-Nivrak committed:
Even more failsafes

Git-Nivrak committed:
shooting in the dark out here

Git-Nivrak committed:
Perhaps this fixes it

Git-Nivrak committed:
Update role_authority.dm

Git-Nivrak committed:
I promise it works now

Git-Nivrak committed:
Might as well fix that

Git-Nivrak committed:
Merge pull request #34 from thwompa/canteenfix

canteen access and ow

Git-Nivrak committed:
revert that

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

thwompa committed:
much awesomer canteen

Git-Nivrak committed:
Merge pull request #36 from thwompa/canteenLINT

Canteen lint EXTRA MONEY EDITION

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Update USS_Almayer.dmm

thwompa committed:
Update USS_Almayer.dmm

thwompa committed:
review

thwompa committed:
fixes

Git-Nivrak committed:
Merge pull request #37 from thwompa/canteen-access

access for preps

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
This was a bad idea I think

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Can't have nice things

Git-Nivrak committed:
Fix non multiz ladders

thwompa committed:
port

Git-Nivrak committed:
Merge pull request #38 from thwompa/kitcheng

port kitchen

Git-Nivrak committed:
ladders again

Git-Nivrak committed:
and cic map again

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Deduplicate that

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
mug rack

Git-Nivrak committed:
Fixup maps in TGM format

abeedd8965c8: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Update hybrisa_props.dm

thwompa committed:
canteen tweaks and also removes mre from utility vendor

Git-Nivrak committed:
Merge pull request #40 from thwompa/kitchen-additions

canteen tweaks and also removes mre from utility vendor

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Git-Nivrak committed:
Fixup maps in TGM format

4723ab79c130: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Actually give them the new colors

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Git-Nivrak committed:
Update USS_Almayer.dmm

Git-Nivrak committed:
Update USS_Almayer.dmm

thwompa committed:
misc fixes

thwompa committed:
more tweeks

thwompa committed:
tweeo

Git-Nivrak committed:
Merge pull request #41 from thwompa/doors

misc fixes

Git-Nivrak committed:
Fixup maps in TGM format

eaeb38eb1cce: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Fix this for the tgui say users

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Git-Nivrak committed:
Update USS_Almayer.dmm

DexterDude reviewed (changes requested):
Dark  HUD needs to have the proper frame as its frame is different from all of the other HUDs for some reason. All of the HUD types also lack the squad-based colors for the indicators. Looks like they're currently using echo's colors. Current squad trackers below, alongside what kilo and oscar currently are.
<img width="865" height="116" alt="squadcolors" src="https://github.com/user-attachments/assets/6615217f-99f7-4b1d-8b85-119bd336062a" />
<img width="1148" height="183" alt="kilooscar" src="https://github.com/user-attachments/assets/2e167b9d-d6a6-44a2-92aa-4ed72b0e8948" />



The colors for the squads also should be reflected in the borders of their spawn landmarks like all other squads, as seen in the image below.
<img width="542" height="126" alt="current" src="https://github.com/user-attachments/assets/39c05e4b-1877-4046-8208-9222478052e5" />


Beret, both onmob and object sprites should use different shades, as is it's a bit too difficult to see what squad it is because the contrast between the squad color and the rest of the beret is so low. Bravo/Charlie next to Kilo and Oscar for reference.
<img width="877" height="308" alt="dark2dark" src="https://github.com/user-attachments/assets/0c39ccfc-5ac6-48c1-854b-ddd2224f89ef" />


Radio headsets need to have their squad color visible, you can see other squads do this here.
<img width="551" height="118" alt="squadkeys" src="https://github.com/user-attachments/assets/2f825cbc-1c3a-474d-9de9-dd21c843babd" />

Steelpoint commented on maps/map_files/USS_Almayer/USS_Almayer.dmm:
<img width="586" height="630" alt="green1" src="https://github.com/user-attachments/assets/558c3c18-04d7-4e32-b51b-e28e588a91eb" />

Suggest adding some green tiles to the support room (example mapped in) to reflect a similar style to Alpha/Bravo/Delta.

Steelpoint replied:
<img width="679" height="341" alt="tablegreen2" src="https://github.com/user-attachments/assets/a8c51999-b1b3-4382-a4d2-86d6d22af22e" />

Put something on these tables. Even just some junk is fine. They shouldn't be bare.

Steelpoint replied:
<img width="805" height="672" alt="roguepurp3" src="https://github.com/user-attachments/assets/118bc5f9-d652-4e1c-9767-411244469f17" />

Replace these purple tiles with yellow ones.

Steelpoint reviewed (changes requested):
See below.

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Commit:
deconflict

Commit:
Fixup maps in TGM format

86d5301c32db: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Git-Nivrak committed:
Update constants.ts

SpypigDev reviewed (commented):
review part 1

thwompa committed:
sprite review

thwompa committed:
map review

Git-Nivrak committed:
Merge pull request #42 from thwompa/ass-review

Ass review

Git-Nivrak committed:
review

thwompa committed:
apc fix

Git-Nivrak committed:
Merge pull request #43 from thwompa/ass-review

apc fix

SpypigDev reviewed (commented):
a few more

MistChristmas committed:
LordWhyIsItNotSorted

Git-Nivrak committed:
Merge pull request #44 from MistChristmas/M10

M10 AP + AP Ext -> Platoon1/2 prep vendors

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

hry-gh committed:
Merge remote-tracking branch 'upstream/master' into ass3

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Git-Nivrak committed:
review

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3

Steelpoint reviewed (dismissed):
Mapping LGTM.

Git-Nivrak committed:
Merge remote-tracking branch 'upstream/master' into ass3
